File: five-hud/html/hud.js
window.addEventListener('message', function(event) {
  if (event.data.action === 'updateHUD') {
    document.getElementById('cash').textContent = event.data.cash;
    document.getElementById('bank').textContent = event.data.bank;
    document.getElementById('crypto').textContent = event.data.crypto;
    document.getElementById('jobName').textContent = event.data.jobLabel || 'Sin trabajo';
    document.getElementById('jobGrade').textContent = event.data.jobGradeLabel ? `- ${event.data.jobGradeLabel}` : '';
  }
  if (event.data.action === 'toggleHUD') {
    const hud = document.getElementById('hud');
    hud.style.opacity = hud.style.opacity === '0' ? '1' : '0';
    hud.style.transition = 'opacity 0.3s ease';
  }
});

File: five-hud/client/hud.lua
RegisterNetEvent('fivecore:hud:update', function(cash, bank, crypto, job)
    SendNUIMessage({
        action = 'updateHUD',
        cash   = string.format("%d", cash),
        bank   = string.format("%d", bank),
        crypto = string.format("%.2f", crypto),
        jobLabel = job and Config.Jobs[job.name] and Config.Jobs[job.name].label or 'Sin trabajo',
        jobGradeLabel = job and Config.Jobs[job.name] and Config.Jobs[job.name].grades[job.grade] and Config.Jobs[job.name].grades[job.grade].name or ''
    })
end)

AddEventHandler('onClientMapStart', function()
    TriggerServerEvent('fivecore:hud:requestUpdate')
end)

-- Toggle HUD con F7
CreateThread(function()
    while true do
        Wait(0)
        if IsControlJustPressed(0, 168) then
            SendNUIMessage({ action = 'toggleHUD' })
        end
    end
end)

